src/InterpreterExpressions.ml

```diff
@@ -108,52 +110,27 @@ let access_rplace_reorganize (config : C.config) (expand_prim_copy : bool)
     ctx
 
 (** Convert an operand constant operand value to a typed value *)
-let rec operand_constant_value_to_typed_value (ctx : C.eval_ctx) (ty : T.ety)
-    (cv : E.operand_constant_value) : V.typed_value =
+let typecheck_constant_value (ty : T.ety)
```

Comment:
I renamed it "constant_to_typed_value".
However in general for the boolean, IMO the issue is that we're potentially loosing error/assertion information (in this case it doesn't matter much because the error message is not great / parametrized anyway ...).

Reply:
Of course, `typecheck_...` could return detailed error information, but what I meant is that `typecheck_...` sounds like a check, not like a function which constructs a value :)

Reply:
Okay yes, I agree ^^

Reply:
Cool :)